### PR TITLE
README fix, update Table Of Contents, fix and better memory backend s…

### PR DESCRIPTION
**Pull request by chyezh**
### Background

I found that README format was dirty due to some conflicts.

### Changes

README fix, update Table Of Contents, fix and better memory backend setup guide.

### Documentation

no code was changed.

### Test Plan

no code was changed.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,19 +48,20 @@ Your support is greatly appreciated
...
     - [🌲 Pinecone API Key Setup](#-pinecone-api-key-setup)
     - [Milvus Setup](#milvus-setup)
-    - [Setting up environment variables](#setting-up-environment-variables-1)
-  - [Setting Your Cache Type](#setting-your-cache-type)
   - [View Memory Usage](#view-memory-usage)
   - [🧠 Memory pre-seeding](#-memory-pre-seeding)
   - [💀 Continuous Mode ⚠️](#-continuous-mode-️)
   - [GPT3.5 ONLY Mode](#gpt35-only-mode)
   - [🖼 Image Generation](#-image-generation)
+  - [Selenium](#selenium)
```
Comment (chyezh) [inline README.md:64]: update table of content

File: README.md [hunk 2]
```diff
@@ -254,7 +255,18 @@ export GOOGLE_API_KEY="YOUR_GOOGLE_API_KEY"
 export CUSTOM_SEARCH_ENGINE_ID="YOUR_CUSTOM_SEARCH_ENGINE_ID"
 ```
 
-## Redis Setup
+## Memory Backend Setup
```
Comment (chyezh) [inline README.md:266]: put ‘Setting Your Cache Type’ section here and fix title hierarchy

File: README.md [hunk 3]
```diff
@@ -347,7 +347,8 @@ To switch to either, change the `MEMORY_BACKEND` env variable to the value that
 
 ## 🧠 Memory pre-seeding
 
-# python autogpt/data_ingestion.py -h
```
Comment (chyezh) [inline README.md:358]: it's a code block, not a title.

File: README.md [hunk 4]
```diff
@@ -293,20 +305,6 @@ Pinecone enables the storage of vast amounts of vector-based memory, allowing fo
 2. Choose the `Starter` plan to avoid being charged.
 3. Find your API key and region under the default project in the left sidebar.
 
-### Milvus Setup
```
Comment (chyezh) [inline README.md:304]: pinecone setup section was splitted by milvus setup.
move the milvus setup down.

**Review by richbeales — APPROVED** [review]

**Comment by richbeales** [comment]
This now has conflicts.

**Comment by chyezh** [comment]
> This now has conflicts.

conflicts fixed.

**Comment by BillSchumacher** [comment]
This looks a lot better now, thanks.

**Review by BillSchumacher — APPROVED** [review]

**Pull request merged**